tools/introspection/source/introspection_app.cpp

```diff
@@ -228,14 +228,17 @@ void IntrospectionApp::waitForUserInput(int32_t timeoutMs)
     fileDesc.fd = STDIN_FILENO;
     fileDesc.events = POLLIN;
     constexpr size_t nFileDesc = 1u;
-    /// @todo iox-#1692 Wrap kernel calls with posixCall
-    int32_t eventCount = poll(&fileDesc, nFileDesc, timeoutMs);
+
+    auto eventCount = iox::posix::posixCall(poll)(&fileDesc, nFileDesc, timeoutMs).failureReturnValue(-1).evaluate();
```

Comment:
You could also use here our awesome `and_then` API :smile: and rewrite the code like this:

```cxx
iox::posix::posixCall(poll)(&fileDesc, nFileDesc, timeoutMs).failureReturnValue(-1).evaluate()
  .and_then([&](auto eventCount){
    if ( eventCount == nFileDesc && fileDesc.revents == POLLIN ) {
      updateDisplayYX();
      refreshTerminal();
    }
  });
```